### PR TITLE
Provide means to pass kwargs to policy in PPO2

### PR DESCRIPTION
**Pull request by huvar**

### CODE REVIEW

**Comment by araffin** [comment]
Hello,
I'm not sure why this extra PR is needed (because related to #83 ).
Also I see some issues here:
- only PPO2 is supported
- adding kwargs mask what is really pass to the algorithm, I would rather use `partial` , or define a custom policy as in https://stable-baselines.readthedocs.io/en/master/guide/custom_policy.html

My main concern with passing kwargs as you did it is that if the use pass wrong kwargs, it will still work... (e.g. passing `learnning_rate` instead of `learning_rate`)

**Comment by huvar** [comment]
> Hello,
> I'm not sure why this extra PR is needed (because related to #83 ).
> Also I see some issues here:
> 
>     * only PPO2 is supported
> 
>     * adding kwargs mask what is really pass to the algorithm, I would rather use `partial` , or define a custom policy as in https://stable-baselines.readthedocs.io/en/master/guide/custom_policy.html
> 
> 
> My main concern with passing kwargs as you did it is that if the use pass wrong kwargs, it will still work... (e.g. passing `learnning_rate` instead of `learning_rate`)

I was inspired by what is done in baselines where a **network_kargs is used. It is useful to control the hyper-parameters. The problem with CustomPolicies (correct me if I am wrong) is that since the object is instantiated from inside the model, we cannot control the network hyper-parameters and I will need a CostumPolicy for each difference in hyper-parameters.

Regarding the support only for PPO2, I think it should not be difficult to extend to other algorithms.

**Comment by araffin** [comment]
>The problem with CustomPolicies (correct me if I am wrong) is that since the object is instantiated from inside the model

I think you can have better way of changing the kwargs using python `partial`  (see https://docs.python.org/3.6/library/functools.html#functools.partial)
How it is used for instance with DQN: https://github.com/hill-a/stable-baselines/blob/a02ead2244bce0c3527a7ddad8349b03b2433317/stable_baselines/deepq/experiments/run_atari.py#L30

**Comment by araffin** [comment]
Closing this PR in favor of #162 that adds kwargs for all algos + tests + documentation, after discussion in #89
